include/cuco/detail/static_set/static_set_ref.inl

```diff
@@ -148,14 +148,14 @@ class operator_impl<op::insert_tag,
   /**
    * @brief Inserts an element.
    *
-   * @tparam Value Input type which is implicitly convertible to 'value_type'
+   * @tparam ProbeKey Input key type which is convertible to 'key_type'
    *
    * @param value The element to insert
    *
    * @return True if the given element is successfully inserted
    */
-  template <typename Value>
-  __device__ bool insert(Value const& value) noexcept
```

Comment:
I was using `value_type` mainly to follow STL convention https://en.cppreference.com/w/cpp/container/unordered_set/insert. They probably do this for a reason. The same for other insert-like functions.

Reply:
Yeah, it's a feature that isn't included in the STL containers, so this is new terrain. Do you think the name `Value` is too vague?

Reply:
The point is your input type change. STL requires it as `value_type` but you change the input type requirement as convertible to `key_type`

Reply:
> convertible to key_type
To `value_type`.

Having this not fixed to the native type of the container is a requirement for heterogeneous insert. So we can either follow STL or have that feature. Maybe I'm not getting the point. 🙃

Reply:
Thanks for the offline clearification. Addressed in [f933f72](https://github.com/NVIDIA/cuCollections/pull/387/commits/f933f728d8477461233f5def165912cd0c6f8233)